django/thunderstore/frontend/api/experimental/serializers/views.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework import serializers
+
+from thunderstore.community.models import PackageCategory
+
+
+class CommunityCardSerializer(serializers.Serializer):
+    """
+    Data shown on "CommunityCard" component on frontend.
+    """
+
+    download_count = serializers.IntegerField(min_value=0)
+    bg_image_src = serializers.CharField(allow_null=True)
+    identifier = serializers.CharField(max_length=256)
+    package_count = serializers.IntegerField(min_value=0)
+    name = serializers.CharField(max_length=256)
+
+
+class PackageCategorySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = PackageCategory
+        fields = ["name", "slug"]
+
+
+class PackageVersionSerializer(serializers.Serializer):
+    """
+    Data shown on "PackageVersions" component on frontend.
+    """
+
+    date_created = serializers.DateTimeField()
+    download_count = serializers.IntegerField(min_value=0)
+    download_url = serializers.CharField()
+    install_url = serializers.CharField()
+    version_number = serializers.CharField(max_length=16)
+
+
+class PackageCardSerializer(serializers.Serializer):
+    """
+    Data shown on "PackageCard" component on frontend.
+    """
+
+    categories = PackageCategorySerializer(many=True)
+    community_name = serializers.CharField(max_length=256)
+    community_identifier = serializers.CharField(max_length=256)
+    deprecated = serializers.BooleanField()
+    description = serializers.CharField(max_length=256)
+    download_count = serializers.IntegerField(min_value=0)
+    image_src = serializers.CharField(allow_null=True)
+    last_updated = serializers.DateTimeField()
+    nsfw = serializers.BooleanField()
+    package_name = serializers.CharField(max_length=128)
+    pinned = serializers.BooleanField()
+    rating_score = serializers.IntegerField(min_value=0)
+    team_name = serializers.CharField(max_length=64)
+
+
+class PackageDependencySerializer(serializers.Serializer):
+    """
+    Data shown on "PackageRequirements" component on frontend.
+    """
+
+    community_name = serializers.CharField(max_length=256)
+    community_identifier = serializers.CharField(max_length=256)
+    description = serializers.CharField(max_length=256)
+    image_src = serializers.CharField(allow_null=True)
+    package_name = serializers.CharField(max_length=128)
+    version_number = serializers.CharField(max_length=16)
+
+
+class CommunityPackageListSerializer(serializers.Serializer):
+    """
+    Data shown on Community's package list view on frontend.
```

Comment:
Are they useful? Well, the codebase for backend is not bite-sized. It can be difficult to find what you're looking for, and it can be difficult to deduce whether what you have found is what you were looking for. Let's say a CLI developer stumbles upon this serializer. Maybe the docstring tells them that this frontend stuff probably isn't what they're looking for? 🤷🏻

That being said, maybe the docstrings in this file don't add that much more value compared to the class name and file path (which tells this is frontend stuff). But maybe it's in more easily digested format? Anyways, if we feel the docstrings don't currently add value, I'd rather consider supplementing them rather than removing them.

About the comment getting stale: since the serializer is used for one specific view in frontend, probably only way it getting stale is that the view is removed from the frontend, meaning that would make the code itself also stale, which renders the issue of having stale comments irrelevant in comparison. (Alternatively it could get stale if the same serializer is used in multiple views, but that's less likely with our "fetch all you need and only things you need for this specific view" approach.)

Reply:
I suppose what we should do is consider what the goals of docstrings we write are and when to write them. As you mentioned, in this case what the docstring documents is _mostly_ self-evident from the class & module path/names. My primary concern is that docstrings aren't really tied to any static analysis, which tends to make them outdated, although I do see your point as well. Really this comes back to documentation practices & maintenance process.

In short, we probably don't need to do anything about the docstrings here and can keep them as is, but we might want to stop to consider how to handle them in the future.